feat: Configure command

fukusuket opened the pull request:
Closed #12
- In this PR, only the configuration for YamatoSecurity is implemented. ASD and Microsoft will be implemented in separate PRs.

## Integraion-Test
- https://github.com/Yamato-Security/WELA/actions/runs/18625879984
  - added configure command test case
    - https://github.com/Yamato-Security/WELA/actions/runs/18625879984/job/53103882202

I’d appreciate it if you could check it when you have time🙏

fukusuket committed:
feat: add ConfigureAuditSettings function to manage audit log configurations

fukusuket committed:
feat: enhance WELA.ps1 to configure event logs and audit policies with improved error handling

fukusuket committed:
feat: add configure command execution to check-audit.yml for PowerShell 5.1

fukusuket committed:
fix: correct typo in check-audit.yml for configure command execution

fukusuket commented:
## Before
<img width="1158" height="352" alt="スクリーンショット 2025-10-19 14 07 24" src="https://github.com/user-attachments/assets/7a10604f-bfd6-486c-81d1-92c0f72302ce" />

## Run configure
<img width="1710" height="1107" alt="スクリーンショット 2025-10-19 14 09 30" src="https://github.com/user-attachments/assets/7cc604ed-87f4-48a6-8d9b-0caa981fbd5c" />

## After
<img width="1710" height="1107" alt="スクリーンショット 2025-10-19 14 09 54" src="https://github.com/user-attachments/assets/3108d247-9a7f-48a5-ba1f-4c03c534afda" />

<img width="1710" height="1107" alt="スクリーンショット 2025-10-19 14 10 43" src="https://github.com/user-attachments/assets/83c8ecb8-1771-46da-99c6-a1ce2dfa2f22" />

fukusuket committed:
feat: enhance audit-settings gui

fukusuket commented:
<img width="1710" height="1107" alt="スクリーンショット 2025-10-19 17 28 03" src="https://github.com/user-attachments/assets/f0ffec45-21b0-4cc1-8d46-3bd5d8267037" />

@YamatoSecurity 
Since it’s preferable to work without having to install tools like TimelineExplorer, I modified the OutType GUI so that it can display three CSV files at the same time. What do you think?(This feature is unrelated to the configure command😅)

fukusuket committed:
feat: update WELA.ps1 to display audit results in GUI format

YamatoSecurity commented:
@fukusuket I like the three OutType GUI popups!
For the configure command, if the user just types `WELA.ps1 configure` can you specify the list of possible baseline settings instead of automatically configuring it as some people might just type `configure` but get surprised when WELA automatically starts changing things.
Also, it would be nice to add a manual (default) configuration as well as an automatic one.
Now it just does everything automatically.

Example: `WELA.ps1 configure -baseline YamatoSecurity` will ask the user every time on whether they want to change their settings.
Ex: `Your current setting is xxxx. Do you want to change it to xxxx? (Y/n)`

If the user adds the `-Auto` option (`WELA.ps1 configure -baseline YamatoSecurity -Auto`, then we can automatically configure everything like it does now.

What do you think?

fukusuket committed:
feat: add auto-configuration option to ConfigureAuditSettings in WELA.ps1

fukusuket committed:
feat: add command execution for auditpol and improve module logging prompts in WELA.ps1

fukusuket committed:
fix: update log size message from 1 GB to 1024 MB in WELA.ps1

fukusuket committed:
fix: update log size message from 1 GB to 1024 MB in WELA.ps1

fukusuket committed:
fix: set default value for Auto parameter in ConfigureAuditSettings function

fukusuket committed:
fix: remove default value for Auto parameter in ConfigureAuditSettings function

fukusuket committed:
fix: update parameter invocation for ConfigureAuditSettings function

fukusuket committed:
fix: update ConfigureAuditSettings function to accept Auto parameter and add new audit policy configuration script

fukusuket commented:
@YamatoSecurity 
Thank you for checking! I changed the default behavior to prompt for Y/N input, and added the -Auto option!
<img width="1710" height="1107" alt="スクリーンショット 2025-10-22 0 13 57" src="https://github.com/user-attachments/assets/f12f06a7-b3a0-42fa-abee-708f3fae2a07" />

<img width="1710" height="1107" alt="スクリーンショット 2025-10-22 0 30 57" src="https://github.com/user-attachments/assets/7c5faa37-3ab6-48a8-833d-45adf407841e" />

fukusuket committed:
fix: update check-audit.yml to include -Auto parameter in configure command

YamatoSecurity commented:
@fukusuket Thanks! I checked that the manual confirmation is working! 
Is it possible to get a list of baselines (ex: `YamatoSecurity`) and the help menu when just `configure` is specified with no other arguments?

Ex:
```
You need to specify a baseline. The following baselines are available:
  * Microsoft
  * YamatoSecurity

Examples: 
./WELA.ps1 configure -Baseline YamatoSecurity
./WELA.ps1 configure -Baseline YamatoSecurity -Auto
```

fukusuket committed:
fix: update WELA.ps1 to handle empty Baseline parameter and provide usage examples for configure command

fukusuket committed:
fix: add Baseline parameter to WELA.ps1 for enhanced configuration options

fukusuket committed:
fix: update check-audit.yml to include Baseline parameter in configure command and add new audit policy configuration script

fukusuket commented:
@YamatoSecurity Thank you for checking! Yes, I updated it!
<img width="1710" height="1107" alt="スクリーンショット 2025-10-22 23 11 17" src="https://github.com/user-attachments/assets/c7d1ee23-8852-4274-82af-6479e2bd163e" />

YamatoSecurity committed:
update changelog

YamatoSecurity reviewed (approved):
@fukusuket Perfect! Thanks so much!!
